subprojects/packagefiles/eyalroz-printf/printf-6.2.0/meson.build

```
@@ -0,0 +1,151 @@
+cc = meson.get_compiler('c')
+## Checks related to the 'j', 'z' and 't' size modifiers
+sizeof_long = cc.sizeof('long')
+sizeof_long_long = cc.sizeof('long long')
+acceptable_jzt_type_sizes = [sizeof_long, sizeof_long_long]
+types = ['intmax_t', 'size_t', 'ptrdiff_t']
+foreach type : types
+  type_size = cc.sizeof(
+    type,
+    prefix: ['#include <stdint.h>'],
+  )
+  if type_size not in acceptable_jzt_type_sizes
+    error(
+      f'sizeof(@type@) is @type_size@, which is neither sizeof(long) (@sizeof_long@) nor sizeof(long long) (@sizeof_long_long@). Please contact the library maintainers with your platform details.',
+    )
+  endif
+endforeach
+
+private_c_args = []
+public_c_args = []
+lib_config = {}
+if cc.get_argument_syntax() == 'gcc'
+  private_c_args += ['-Wall', '-Wextra', '-pedantic', '-Wconversion']
+elif cc.get_argument_syntax() == 'msvc'
+  private_c_args += '/W4'
+endif
```

Maybe just
```suggestion
private_c_args = cc.get_supported_arguments('-Wall', '-Wextra', '-pedantic', '-Wconversion', '/W4')
```

Also, for the record, these are pretty aggressive warning levels for use by the general public.  (It makes sense for project maintainers to use higher levels, but.)

---

The more general issue here is that `meson setup -Dwarning_level=3` obsoletes this. If you just want the default to be to have all these warnings then it can be set in `project(default_options: ...)`.

If passing it to specific targets, there is still `override_options:`. And you can avoid doing any compiler checks or ifelse ladders.